### ISSUE TITLE
Cannot login to Oscar-3.0.40 after upgrade from v1.0.2 to v1.2.0

### ISSUE THREAD

**Issue by dalehenrich** [ericwinger/Jade#195]
This is the error encountered when you try to login using Oscar-3.0.40 after a fresh v1.0.2 install followed by a login using Jadeite-Alpha1.0.2 and then an upgrade to v1.2.0:
![image](https://user-images.githubusercontent.com/623951/45963286-b0ce8b00-bfd7-11e8-9aab-abf83408b7f5.png)
If you skip the Jadeite-Alpha1.0.2 step above, you get the same error reported in GemTalk/Rowan#355:
![image](https://user-images.githubusercontent.com/623951/45934975-6e1a9d80-bf5a-11e8-8bb9-b5b925e4786f.png)

**Comment by dalehenrich** [comment]
Turns out the this particular issue is cause by an incorrect Rowan update script. The following script properly updates Rowan from v1.0.2:
```
#---
run
  "update and load Rowan projects"
  Rowan projectTools load
    loadProjectNamed: 'Cypress';
    loadProjectNamed: 'STON';
    loadProjectNamed: 'Tonel';
    loadProjectNamed: 'Rowan';
    yourself
%
commit
run
    "For update from v1.0.2. Create the Rowan-JadeServer package and adopt the JadeServer classes into the package"
    | jadeServerClass projectDefinition jadeServerProjectName |
    jadeServerClass := UserGlobals at: #JadeServer ifAbsent: [ ^ 'No JadeServer class found' ].
    (jadeServerProjectName := jadeServerClass rowanProjectName) = '(NONE)'
        ifFalse: [ ^ 'JadeServer project name: ',  jadeServerProjectName printString ].
    projectDefinition := (Rowan image loadedProjectNamed: 'Rowan') asDefinition.
    projectDefinition addPackageNamed: 'Rowan-JadeServer'.
    Rowan projectTools load
        loadProjectDefinition: projectDefinition.
    GsFile gciLogServer: 'Created Rowan-JadeServer package'.
    "Adopt the JadeServer classes into the Rowan-JadeServer package"
    {
        'JadeServer64bit3x' .
        'JadeServer' .
        'JadeServer64bit24' .
        'JadeServer64bit' .
        'JadeServer64bit32' .
    } do: [:className |
        Rowan packageTools adopt
            adoptClassNamed: className
            intoPackageNamed: 'Rowan-JadeServer'
        ].
%
run
    "For update from v1.0.2. Do the full load with the current list of group names ('jadeServer' being the most important"
    Rowan projectTools load
        loadProjectNamed: 'Rowan'
        withConfigurations: #( 'Load' )
        groupNames: #( 'core' 'tests' 'deprecated' 'jadeServer' )
%
commit
#---
```

**Comment by dalehenrich** [comment]
Reopening so that we can have regression tests for incorrect and missing JadeServer classes
- [ ] jadeite alpha2.0.1 run agains Oscar stone
- [ ] missing JadeServer classes

**Comment by dalehenrich** [comment]
Here are four stones running on **foos** that should exhibit the reported issue `MNU updateFromSton::
```
	Status       Version    Owner    Pid   Port   Started     Type       Name
	-------     --------- --------- ----- ----- ------------ ------      ----
	exists      3.2.15    dhenrich   7723 44326 Oct 05 15:01 Netldi      jadeite_195_v110_alpha201_3215_ldi
	exists      3.2.15    dhenrich   7796 43086 Oct 05 15:01 Netldi      jadeite_195_v102_alpha201_3215_B_ldi
	exists      3.2.15    dhenrich   7822 33720 Oct 05 15:02 Netldi      jadeite_195_v102_alpha201_3215_ldi
	exists      3.2.15    dhenrich   7937 43471 Oct 05 15:06 Netldi      jadeite_195_v12x_alpha201_3215_ldi

```
I cannot test directly because my windows vm cannot see foos correctly ... login as *SystemUser* and Bob's your uncle

**Comment by dalehenrich** [comment]
error when connecting to jadeite_195_v110_alpha201_3215 stone:
![image](https://user-images.githubusercontent.com/623951/46553054-56fc7980-c891-11e8-8b0d-86137f7973d3.png)

error when connecting to jadeite_195_v102_alpha201_3215:
![image](https://user-images.githubusercontent.com/623951/46553156-a347b980-c891-11e8-9aa1-009940821abf.png)

error connection to jadeite_195_v12x_alpha201_3215:
![image](https://user-images.githubusercontent.com/623951/46555268-97f78c80-c897-11e8-94af-b4ce22b43fd7.png)

error connection to jadeite_195_v102_alpha201_3215_B:
![image](https://user-images.githubusercontent.com/623951/46556557-dc852700-c89b-11e8-950a-c9257df16d1f.png)

**Comment by dalehenrich** [comment]
@ericwinger I've got several stones with a variety of login errors .. that should be enough for you to put together some test cases for the login problems ... I assume that you will be sending messages to JadeServer which will be sufficient for the other case I don't have a stone for ...

**Comment by ericwinger** [comment]
@dalehenrich. Sounds good, just send me the necessary login info.

**Comment by dalehenrich** [comment]
@ericwinger the login info is [here](https://github.com/GemTalk/Jadeite/issues/195#issuecomment-427454541)

**Comment by ericwinger** [comment]
@dalehenrich I'll take a look.

**Comment by ericwinger** [comment]
Results I'm seeing. Not yet seeing the bug your seeing. Perhaps I'm logging in incorrectly. 
SystemUser. Login information is below. 
Is the netldi in guest mode, @dalehenrich? 

![image](https://user-images.githubusercontent.com/36212516/46560482-cc277900-c8a8-11e8-88b5-4652b4131964.png)
![image](https://user-images.githubusercontent.com/36212516/46560603-3b04d200-c8a9-11e8-86f7-508c63d88f4c.png)


Login with: 
exists      3.2.15    dhenrich   4835 36085 Oct 05 10:57 Netldi      jadeite_195_v110_alpha201_3215_ldi
![image](https://user-images.githubusercontent.com/36212516/46560226-c41b0980-c8a7-11e8-90a9-8c0691a4e0fe.png)

Login with: 
exists      3.2.15    dhenrich   4575 45492 Oct 05 10:56 Netldi      jadeite_195_v102_alpha201_3215_ldi
![image](https://user-images.githubusercontent.com/36212516/46560349-40ade800-c8a8-11e8-9268-eb4af8e1eb03.png)

Login with: 
exists      3.2.15    dhenrich   7113 33740 Oct 05 12:34 Netldi    jadeite_195_v102_alpha201_3215_B_ldi
![image](https://user-images.githubusercontent.com/36212516/46560392-781c9480-c8a8-11e8-9fd7-5ea5e6f52bf0.png)

Login with: 
exists      3.2.15    dhenrich   6038 39997 Oct 05 12:03 Netldi      jadeite_195_v12x_alpha201_3215_ldi
![image](https://user-images.githubusercontent.com/36212516/46560448-aa2df680-c8a8-11e8-8d98-d72a8880fb33.png)

**Comment by dalehenrich** [comment]
for one thing the stone is running on 'foos', not 'localhost' ... I am not able to test the stones on foos (as I mentiioned) because my windows vm is not aware of the vpn network ... I'll cycle the netldi on all of the stones to make sure they are picking up the ROWN_PROJECTS_HOME env ... then you'll have to let me know what else breaks ... I tested the extents on my laptop, but it's not clear that you can get to the stones on my laptop when I'm not on the wired network ...

**Comment by dalehenrich** [comment]
@ericwinger, I recycled the netldis so [all of the port numbers have changed](https://github.com/GemTalk/Jadeite/issues/195#issuecomment-427454541) .. so they're ready to try again ...

**Comment by ericwinger** [comment]
Using the new port numbers, I see this message trying to connect to all four stones:

![image](https://user-images.githubusercontent.com/36212516/46622390-375a9080-cadf-11e8-96d7-9422f5658719.png)

Here is one set of parameters to ensure I made the necessary adjustments

![image](https://user-images.githubusercontent.com/36212516/46623040-43dfe880-cae1-11e8-8fb4-fd77a3959e82.png)

![image](https://user-images.githubusercontent.com/36212516/46622381-3164af80-cadf-11e8-97b4-7dae7a87bd75.png)

**Comment by ericwinger** [comment]
Got logged in. Had a ``_ldi`` at the end of the stone name. 

After discussion with @dalehenrich, looks like we need a server side suite of tests that simulate client calls. If nothing else a few tests that touch JadeServer in different ways would suffice.

**Comment by ericwinger** [comment]
I've added a new test class JadeServerTest (Rowan-Services-Tests package) with several tests that look at JadeServer from within the server to ensure it's reachable and usable. All the client tests (run in Dolphin) depend on an instance of JadeServer being available so implicitly they all test JadeServer. 

Will be available in 3.0.41.
